src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,190 @@
+package duke.util;
+
+/**
+ * Parses the commands given by a user.
+ */
+public class Parser {
+    /**
+     * Parses a done command.
+     *
+     * @param cmd Command to be processed.
+     * @return Index of task to be marked done.
+     */
+    public static int parseDoneCmd(String cmd) {
+        String[] a = cmd.split(" ");
+        return Integer.parseInt(a[1]);
+    }
+
+    /**
+     * Checks if command is bye.
+     *
+     * @param cmd Command to be processed.
+     * @return Whether is bye command.
+     */
+    public static boolean isNotBye(String cmd) {
+        return !cmd.equals("bye");
+    }
+
+    /**
+     * Checks if command is list.
+     *
+     * @param cmd Command to be processed.
+     * @return Whether is list command.
+     */
+    public static boolean isList(String cmd) {
+        return cmd.equals("list");
+    }
+
+    /**
+     * Checks if command is done.
+     *
+     * @param cmd Command to be processed.
+     * @return Whether is done command.
+     */
+    public static boolean isDone(String cmd) {
+        return cmd.matches("^done [0-9]+$");
+    }
+
+    /**
+     * Checks if command is a valid task.
+     *
+     * @param cmd Command to be processed.
+     * @return Whether task is valid.
+     */
+    public static boolean isValidTask(String cmd) {
+        return cmd.matches("^todo .*$")
+                || cmd.matches("^deadline .* /by .*$")
+                || cmd.matches("^event .* /at .*$");
+    }
+
+    /**
+     * Checks if command is missing description.
+     *
+     * @param cmd Command to be processed.
+     * @return Whether command is missing argument.
+     */
+    public static boolean isMissingArg(String cmd) {
+        String[] a = cmd.split(" ", 2);
+        return a[1].matches(" *");
+    }
+
+    /**
+     * Gets the task name in a command.
+     *
+     * @param cmd Command to be processed.
+     * @return Name of task.
+     */
+    public static String getTaskName(String cmd) {
+        String[] a = cmd.split(" ", 2);
+        return a[0];
+    }
+
+    /**
+     * Gets the task description in a command.
+     *
+     * @param cmd Command to be processed.
+     * @return Description of task.
+     */
+    public static String getDesc(String cmd) {
+        String[] a = cmd.split(" ", 2);
+        return a[1];
+    }
+
+    /**
+     * Gets the deadline information in a deadline description.
+     *
+     * @param desc Command description to be processed.
+     * @return Deadline information.
+     */
+    public static String getDlInfo(String desc) {
```

Comment:
`getDlInfor` Perhaps Dl might raise uncertainty for it's meaning in the method header. Spelling the full `getDeadlineInfo` might help to reduce uncertainty.